test/unit/test_host.rb

```diff
@@ -65,8 +65,14 @@ def test_assert_hosts_hash_equally
     end
 
     def test_assert_hosts_compare_equal
-      assert Host.new('example.com').eql? Host.new('example.com')
-      assert Host.new('example.com').equal? Host.new('example.com')
+      h1 = Host.new('first_host')
+      h2 = Host.new('second_host')
+      h1.expects(:hash).times(3).returns('example.com')
```

Comment:
Hi @robd.  I don't know if you saw my note on #275 that explained my approach, but basically I'm stating this test is testing too much, basically testing that the initializer is deterministic and obfuscating the reality of what makes two hosts equal.  With the stubs it's purposefully ignorant of the "real life" details and only testing what it's meant to, the definition of equality between hosts.

If you still don't agree I'll be happy to change the test to what you describe. Let me know.